src/main/cpp/src/parse_uri.cu

```diff
@@ -306,6 +776,9 @@ std::unique_ptr<column> parse_uri_to_protocol(strings_column_view const& input,
   auto offsets_column = make_numeric_column(
     data_type{type_to_id<size_type>()}, offset_count, mask_state::UNALLOCATED, stream, mr);
 
+  // build src offsets buffer
+  auto src_offsets = rmm::device_buffer{strings_count * sizeof(size_type), stream};
```

Comment:
Why not `device_uvector<size_type>`?

Reply:
This is simply a temporary buffer of data and is used as such. It could be a `device_uvector`, but we only ever use '.data()`. I'm not against making it a `device_uvector`, but it doesn't buy us much outside of not using `sizeof` directly in our code.